src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -220,30 +201,10 @@ private void Disconnect()
         /// </summary>
         internal NetworkAddress FindPeerToConnectTo()
         {
-            int groupFail = 0;
-
-            NetworkAddress peer = null;
-
-            while (groupFail < 50 && !this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
-            {
-                peer = this.peerAddressManager.SelectPeerToConnectTo(this.peerIntroductionType);
-                if (peer == null)
-                    break;
-
-                if (!peer.Endpoint.Address.IsValid())
-                    continue;
-
-                bool peerExistsInGroup = this.RelatedPeerConnector.GlobalConnectedNodes().Any(a => this.groupSelector(a).SequenceEqual(this.groupSelector(peer.Endpoint)));
-                if (peerExistsInGroup)
-                {
-                    groupFail++;
-                    continue;
-                }
-
-                break;
-            }
-
-            return peer;
+            NetworkAddress peer = this.peerAddressManager.SelectPeerToConnectTo(this.peerIntroductionType);
```

Comment:
hmm where is the code that replaces the GroupConnector?

Reply:
I would suggest to pass ConnectionMaanger as a parent to PeerConenctor (or even only pass the PeerCollection in the constructor, that way you can check a peer is not already connected.